SoomlaiOSProfile/GameServicesController.m

```diff
@@ -0,0 +1,86 @@
+/*
+ Copyright (C) 2012-2015 Soomla Inc.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import <Reward.h>
+#import "GameServicesController.h"
+#import "IGameServicesProvider.h"
+#import "ProfileEventHandling.h"
+#import "SoomlaUtils.h"
+
+
+@implementation GameServicesController
+
+static NSString* TAG = @"SOOMLA GameServicesController";
+
+- (id)initWithParameters:(NSDictionary *)providerParams {
+    if (self = [super init]) {
+
+        // TODO: Check if providerPkgPrefix can be omitted completely in iOS
+        if (![self loadProvidersWithProtocol:@protocol(IGameServicesProvider) andProviderParams:providerParams]) {
+            NSString* msg = @"You don't have a IGameServicesProvider service attached. \
+                            Decide which IAuthProvider you want, and add its static libraries \
+                            and headers to the target's search path.";
+            LogDebug(TAG, msg);
+        } else {
+
+        }
+    }
+
+    return self;
+}
+
+- (id)initWithoutLoadingProviders {
+    if (self = [super init]) {
+    }
+    return self;
+}
+
+-(void)getFriendsListWithProvider:(Provider)provider payload:(NSString *)payload andReward:(Reward *)reward {
+    id<IGameServicesProvider> gsProvider = (id<IGameServicesProvider>)[self getProvider:provider];
+
+    [ProfileEventHandling postGetGameFriendsStarted:provider withPayload:payload];
+    [gsProvider getFriendsListWithSuccess:^(NSArray *userProfiles) {
```

Comment:
@EugeneButusov, support pagination param in signatures anyway. (Actually, if you use GetContacts, it's not relevant).